### PR TITLE
Implement PackageJSON test case

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
*Adds Bayesian test for JSON-based projects
*Refactors BayesianPomXml test case

PR depends on https://github.com/redhat-developer/rh-che/pull/1105 - MUST BE MERGED FIRST

### What issues does this PR fix or reference?
https://app.zenhub.com/workspaces/osio-che-team-59411dba86156437e0727c12/issues/redhat-developer/che-functional-tests/294

### How have you tested this PR?
manual testing against minishift and prod-preview

### CODE REVIEW

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    editor.closeAllTabs();
+  }
+
+  @Test
+  public void createBayesianError() {
+    try {
+      editor.waitTextInToolTipPopup(ERROR_MESSAGE);
+    } catch (Exception e) {
+      LOG.error(
+          "Bayesian error not present after adding dependency - known issue for prod-preview.");
```
Comment (Katka92) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: Would you like to show this error even if this fails on prod? I don't see any switch or check if the test runs on prod-preview. Same for ```checkErrorPresentAfterReopenFile``` method.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      return;
+    }
+    LOG.info("Bayesian error message was present after adding dependency.");
+  }
+
+  @Test
```
Comment (Katka92) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: Are you sure this test would ALWAYS run as a seconds one? I would suggest to add ```priority``` here to be sure about an order as far as it depends on ```createBayesianError```. @rhopp are you OK that these methods aren't atomics?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java [hunk 3]
```diff
@@ -21,68 +24,91 @@
...
+  }
+
+  @AfterMethod
+  public void closeFiles() {
+    removeDependency();
+    editor.closeAllTabs();
   }
 
-  @Test(priority = 2)
+  @Test
```
Comment (Katka92) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: Why did you remove priority from here?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java (hunk 1 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: fixed

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java (hunk 2 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: Those tests are re-written to be atomic. the order doesn't matter anymore.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java (hunk 3 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: Both tests were re-written to be atomic, the order therefore doesn't matter
